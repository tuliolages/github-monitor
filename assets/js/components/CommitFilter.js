import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Select from 'react-select'


const CommitFilter = (props) => {
  const { repository, repositories, author, authors } = props

  const repositoryOptions = [
    { label: 'All' },
    ...repositories.map(({ name, id }) => ({ label: name, value: id }))
  ]
  const currentRepositoryOption = (repositoryOptions || []).find(({ value }) => value === repository) || repositoryOptions[0]

  const authorOptions = [
    { label: 'All' },
    ...authors.map(({ author }) => ({ label: author, value: author }))
  ]
  const currentauthorOption = (authorOptions || []).find(({ value }) => value === author) || repositoryOptions[0]

  return (
    <div className="card card-outline-secondary my-4">
      <div className="card-header">
        Filters
      </div>

      <div className="card-body">
        <div className="form-group">
          <div className="form-row">
            <div className="form-group col-md-6">
              <label htmlFor="repositories">Repositories</label>
              <Select
                id="repositories"
                value={currentRepositoryOption}
                onChange={props.onChange('repository')}
                options={repositoryOptions}
              />
            </div>
            <div className="form-group col-md-6">
              <label htmlFor="authors">Authors</label>
              <Select
                id="authors"
                value={currentauthorOption}
                onChange={props.onChange('author')}
                options={authorOptions}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

CommitFilter.propTypes = {
  repository: PropTypes.number,
  repositories: PropTypes.arrayOf(PropTypes.object).isRequired,
  onChange: PropTypes.func.isRequired,
  author: PropTypes.string,
  authors: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default CommitFilter;
