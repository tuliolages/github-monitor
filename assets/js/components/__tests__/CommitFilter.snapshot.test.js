import React from 'react';
import { shallow } from 'enzyme';
import CommitFilter from '../CommitFilter';

test('Render empty filter selects', () => {
  const component = shallow(
    <CommitFilter
      authors={[]}
      repositories={[]}
      onChange={() => { }}
    />
  );
  expect(component).toMatchSnapshot();
});

test('Render with filter select options', () => {
  const component = shallow(
    <CommitFilter
      authors={[{ author: 'author1' }, { author: 'author2' },]}
      repositories={[{ id: 1, name: 'repo1' }, { id: 2, name: 'repo2' },]}
      onChange={() => { }}
    />
  );
  expect(component).toMatchSnapshot();
});

test('Render with filter select options and already selected options', () => {
  const component = shallow(
    <CommitFilter
      author={"author1"}
      authors={[{ author: 'author1' }, { author: 'author2' }]}
      repository={1}
      repositories={[{ id: 1, name: 'repo1' }, { id: 2, name: 'repo2' }]}
      onChange={() => { }}
    />
  );
  expect(component).toMatchSnapshot();
});
