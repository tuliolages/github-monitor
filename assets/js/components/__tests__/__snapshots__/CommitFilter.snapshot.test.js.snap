// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render empty filter selects 1`] = `
<div
  className="card card-outline-secondary my-4"
>
  <div
    className="card-header"
  >
    Filters
  </div>
  <div
    className="card-body"
  >
    <div
      className="form-group"
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="repositories"
          >
            Repositories
          </label>
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            id="repositories"
            options={
              Array [
                Object {
                  "label": "All",
                },
              ]
            }
            value={
              Object {
                "label": "All",
              }
            }
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="authors"
          >
            Authors
          </label>
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            id="authors"
            options={
              Array [
                Object {
                  "label": "All",
                },
              ]
            }
            value={
              Object {
                "label": "All",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render with filter select options 1`] = `
<div
  className="card card-outline-secondary my-4"
>
  <div
    className="card-header"
  >
    Filters
  </div>
  <div
    className="card-body"
  >
    <div
      className="form-group"
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="repositories"
          >
            Repositories
          </label>
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            id="repositories"
            options={
              Array [
                Object {
                  "label": "All",
                },
                Object {
                  "label": "repo1",
                  "value": 1,
                },
                Object {
                  "label": "repo2",
                  "value": 2,
                },
              ]
            }
            value={
              Object {
                "label": "All",
              }
            }
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="authors"
          >
            Authors
          </label>
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            id="authors"
            options={
              Array [
                Object {
                  "label": "All",
                },
                Object {
                  "label": "author1",
                  "value": "author1",
                },
                Object {
                  "label": "author2",
                  "value": "author2",
                },
              ]
            }
            value={
              Object {
                "label": "All",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render with filter select options and already selected options 1`] = `
<div
  className="card card-outline-secondary my-4"
>
  <div
    className="card-header"
  >
    Filters
  </div>
  <div
    className="card-body"
  >
    <div
      className="form-group"
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="repositories"
          >
            Repositories
          </label>
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            id="repositories"
            options={
              Array [
                Object {
                  "label": "All",
                },
                Object {
                  "label": "repo1",
                  "value": 1,
                },
                Object {
                  "label": "repo2",
                  "value": 2,
                },
              ]
            }
            value={
              Object {
                "label": "repo1",
                "value": 1,
              }
            }
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="authors"
          >
            Authors
          </label>
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            id="authors"
            options={
              Array [
                Object {
                  "label": "All",
                },
                Object {
                  "label": "author1",
                  "value": "author1",
                },
                Object {
                  "label": "author2",
                  "value": "author2",
                },
              ]
            }
            value={
              Object {
                "label": "author1",
                "value": "author1",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
